src/web/workers/InputWorker.mjs

```diff
@@ -0,0 +1,1048 @@
+            return;
+        }
+    }
+    self.postMessage({
+        action: "activateLoaderWorker"
+    });
+};
+
+/**
+ * Sends a message to the inputWaiter to terminate a loaderWorker.
+ *
+ * @param {number} id - The ID of the worker to be terminated
+ */
+self.terminateLoaderWorker = function(id) {
+    self.postMessage({
+        action: "terminateLoaderWorker",
+        data: id
+    });
+    // If we still have pending files, spawn a worker
+    if (self.pendingFiles.length > 0) {
+        self.activateLoaderWorker();
+    }
+};
+
+/**
+ * Loads files using LoaderWorkers
+ *
+ * @param {object} filesData
+ * @param {FileList} filesData.files - The list of files to be loaded
+ * @param {number} filesData.activeTab - The active tab in the UI
+ */
+self.loadFiles = function(filesData) {
+    const files = filesData.files;
+    const activeTab = filesData.activeTab;
+    let lastInputNum = -1;
+    const inputNums = [];
+    for (let i = 0; i < files.length; i++) {
+        if (i === 0 && self.getInputValue(activeTab) === "") {
+            self.removeInput({
+                inputNum: activeTab,
+                refreshTabs: false,
+                removeChefWorker: false
+            });
+            lastInputNum = self.addInput(false, "file", {
+                name: files[i].name,
+                size: files[i].size.toLocaleString(),
+                type: files[i].type || "unknown"
+            }, activeTab);
+        } else {
+            lastInputNum = self.addInput(false, "file", {
+                name: files[i].name,
+                size: files[i].size.toLocaleString(),
+                type: files[i].type || "unknown"
+            });
+        }
+        inputNums.push(lastInputNum);
+
+        self.pendingFiles.push({
+            file: files[i],
+            inputNum: lastInputNum
+        });
+    }
+    let max = self.maxWorkers;
+    if (self.pendingFiles.length < self.maxWorkers) max = self.pendingFiles.length;
+
+    // Create loaderWorkers to load the new files
+    for (let i = 0; i < max; i++) {
+        self.activateLoaderWorker();
+    }
+
+    self.getLoadProgress();
+    self.setInput({inputNum: activeTab, silent: false});
+};
+
+/**
+ * Adds an input to the input dictionary
+ *
+ * @param {boolean} [changetab=false] - Whether or not to change to the new input
+ * @param {string} type - Either "string" or "file"
+ * @param {Object} fileData - Contains information about the file to be added to the input (only used when type is "file")
+ * @param {string} fileData.name - The filename of the input being added
+ * @param {number} fileData.size - The file size (in bytes) of the input being added
+ * @param {string} fileData.type - The MIME type of the input being added
+ * @param {number} inputNum - Defaults to auto-incrementing self.currentInputNum
+ */
+self.addInput = function(changeTab=false, type, fileData={name: "unknown", size: "unknown", type: "unknown"}, inputNum = self.currentInputNum++) {
```

Comment:
Might want to lay this line out a bit better with newlines and indentation. Also just double check whether having an expression with `++` in this line will increment it regardless of whether it is used.